File: CSharp/HtmlWalker/Tag.Util/Entity.cs
using HtmlWalker.Core;
using HtmlWalker.Platform;
using HtmlWalker.Tag.Base;
using HtmlWalker.Visitor;

namespace HtmlWalker.Tag.Util
{
    /// <summary>
    ///   Holds an HTML entity
    /// </summary>
    /// <author>Robert R Flores</author>
    public class Entity : Text
    {
        /// <summary>
        ///   Instantiates a new entity adapter
        /// </summary>
        /// <param name="factory">the tag that owns the text</param>
        /// <param name="entity">the entity string</param>
        public Entity(WalkerPlatform.WalkerFactory factory, string entity) : this(factory, factory.CreateEntityNode(entity)) { }

        /// <summary>
        ///   Constructor for housing a existing text
        /// </summary>
        /// <param name="factory">used to produce objects as per the specific platform</param>
        /// <param name="obj">object managed by the adapter</param>
        public Entity(WalkerPlatform.WalkerFactory factory, object obj) : base(factory, obj) { }

        /// <summary>
        ///   Implements <see cref="ITag.Visit(ITagVoidVisitor)"/>
        /// </summary>
        public override void Visit(ITagVoidVisitor visitor) { visitor.Visit(this); }

        /// <summary>
        ///   Implements <see cref="ITag.Visit{T}(ITagGenericVisitor{T})"/>
        /// </summary>
        public override T Visit<T>(ITagGenericVisitor<T> visitor) { return visitor.Visit(this); }

        /// <summary>
        ///  Override <see cref="BaseTag.Name"/> for one that identifies it utility function
        /// </summary>
        public override string Name => "#entity";

        /// <summary>
        ///   Implements <see cref="ITag.TextValue"/>
        /// </summary>
        public override string TextValue
        {
            get => EscHandler.Instance.Escape(Api.GetText(Tag)); set => Api.SetText(Tag, value);
        }
    }
}

File: Java/src/htmlwalker/tag/util/Entity.java
package htmlwalker.tag.util;

import htmlwalker.core.EscHandler;
import htmlwalker.exception.HtmlWalkerException;
import htmlwalker.platform.WalkerFactory;
import htmlwalker.visitor.ITagGenericVisitor;
import htmlwalker.visitor.ITagVoidVisitor;

public class Entity extends Text
{
    public Entity(WalkerFactory factory, String entity) { super(factory, factory.createEntityNode(entity)); }
    
    public Entity(WalkerFactory factory, Object tag) { super(factory, tag); }

    @Override
    public void visit(ITagVoidVisitor visitor) throws HtmlWalkerException { visitor.visit(this); }

    @Override
    public <T> T visit(ITagGenericVisitor<T> visitor) throws HtmlWalkerException { return visitor.visit(this); }
    
    @Override
    public String name() { return "#entity"; }
    
    @Override
    public String getTextValue() { return EscHandler.theInstance.escape(this.api.getText(tag())); }
    
    @Override
    public void setTextValue(String text) { throw new IllegalStateException(); }

}
